backend/src/main/java/com/jobwait/persistence/adapters/PersistenceAdapter.java

```diff
@@ -0,0 +1,7 @@
+package com.jobwait.persistence.adapters;
+
+import java.sql.ResultSet;
+
+abstract public class PersistenceAdapter<A> {
+    public abstract A fromResultSetRow(ResultSet rs) throws AdapterException;
```

Comment:
what is the future intention for this or is this just here to justify PostgresUserAdapter?

Reply:
PersistenceAdapter serves to turn any ResultSet that java.sql gives us into something actually usable (like a class). There will eventually be a `PostgresAnswerAdapter` that extends this, when we want to fetch answers from the DB.

Reply:
ok this is far clearer now, thank you. sounds good. so the flow is really `RequestController` -> `PostgresController` (some sql data here) -> Adapter of choice which adapts data into domain-type classes -> return?

Reply:
yeah pretty much